src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/sequences/general/SubsequenceFunctionIterator.java

```diff
@@ -179,7 +180,7 @@ public void setNextResult() {
                 } else if (_length == -1) { // _length not specified -> take all items until the end
                     _nextResult = _sequenceIterator.next();
                 } else {
-                    throw new SparksoniqRuntimeException(
+                    throw new OurBadException(
                             "Unexpected length value found. Please report the bug with subsequence function iterator."
```

Comment:
I feel that this error message duplicates the 'please report a bug' sentiment. Shall we change it to something along the lines of 'Unexpected length value found in subsequence iterator'?

Reply:
I agree too.